modules/helpers.py

```diff
@@ -69,6 +69,13 @@ def target_creator(command_line_object):
 
         if root.tag.lower() == "nmaprun" and root.attrib.get('scanner') == 'nmap':
             print "Detected nmap xml file\n"
+
+            # command line provided ports
+            # in nmap logic, https ports must also be http 
+            http_ports += command_line_object.add_http_ports
+            http_ports += command_line_object.add_https_ports
```

Comment:
It's related to the nested 'if' logic at 107 and 109 (in the original file) where it only checks if something is 'https' if it has already established that it's 'http', since nmap differentiates between service and tunnel but EyeWitness doesn't.

Could be solved by un-nesting that logic instead.